SleepEngine/Engine/GameTemplate/Game.cpp

```diff
@@ -32,66 +31,45 @@ Game::Game(size_t width, size_t height)
 
     if (m_instance)
     {
-	    LOG_AND_ASSERT_ERROR(false, "Attempt to create second game window! It is forbidden!");
-	    return;
+        LOG_AND_ASSERT_ERROR(false, "Attempt to create second game window! It is forbidden!");
+        return;
     }
     m_instance = this;
     m_renderer = std::make_unique <GameRenderer>();
 }
 
-Game::SceneIDType Game::addScene(SceneIniter initer)
+void Game::addScene(Scene::Initer initer, std::string_view id)
 {
-    auto sceneAndIniter = std::make_pair(Scene(), initer);
-    auto [it, wasInserted] = m_scenes.insert({ m_nextSceneID++, std::move(sceneAndIniter) });
-    assertion(wasInserted, "nextSceneID overflow");
+    auto& scene = m_scenes.emplace_back(std::move(initer), id);
 
     if (m_scenes.size() == 1)
     {
-        m_currentScene = it;
-        initCurrentScene();
+        // first scene
+        scene.init();
+        m_currentScene = m_scenes.begin();
     }
-
-    return it->first;
-}
-
-Game::Scene* Game::findScene(SceneIDType id)
-{
-    auto it = m_scenes.find(id);
-
-    return it != m_scenes.end() ? &it->second.first : nullptr;
-}
-
-Game::Scene const* Game::findScene(SceneIDType id) const
-{
-    return const_cast <Game*>(this)->findScene(id);
 }
 
-bool Game::tryRemoveScene(SceneIDType id)
+void Game::changeActiveSceneTo(std::string_view id)
 {
-    auto it = m_scenes.find(id);
-
-    if (it == m_scenes.end())
+    auto const isSceneFound = [id](auto const& scene)
     {
-        return false;
-    }
-
-    m_scenes.erase(it);
-    return true;
-}
+        return scene.getID() == id;
+    };
 
-void Game::changeScene(SceneIDType id)
-{
-    auto it = m_scenes.find(id);
+    auto const[found, it] = findIf(m_scenes, isSceneFound);
 
-    if (it == m_currentScene)
+    if (!found)
     {
+        LOG_AND_FAIL("No scene found with id: {}", id);
         return;
     }
 
-    m_currentScene->second.first.clear();
-    m_currentScene = it;
+    spdlog::get(EngineLogger)->info("Changing scene '{}' to '{}'.", m_currentScene->getID(), id);
 
-    initCurrentScene();
+    m_entityManager.clear();
+    
+    it->init();
```

Comment:
Oh shit. Found a problem here. I guess we should `forceClear` entity manager here.

Reply:
Or maybe it will be ok, should test it

Reply:
for the first time it will be ok, but only then

Reply:
Fixed